resolwe/flow/tests/test_access_api.py

```diff
@@ -0,0 +1,101 @@
+from resolwe.test import TransactionProcessTestCase, check_installed, with_docker_executor, with_resolwe_api_host
+
+
+class AccessAPIFromExecutorProcessTestCase(TransactionProcessTestCase, LiveServerTestCase):
+
+    def create_collection(self):
+        """Create a test collection that will be accessed via a test process.
+
+        Assign :class:`~django.contrib.auth.models.AnonymousUser`
+        ``view_collection`` permission to the created collection.
+
+        :return: created test collection
+        :rtype: Collection
+
+        """
+        collection = Collection.objects.create(
+            name='collection-foo',
+            contributor=self.contributor,
+        )
+        assign_perm("view_collection", AnonymousUser(), collection)
+        return collection
+
+    def create_process(self):
+        """Create a test process that will query the API for collections.
+
+        Get the list of all collections using ``curl`` to query the Live
+        Resolwe API host.
+
+        :return: created test process
+        :rtype: Process
+
+        """
+        process = Process.objects.create(
+            name='Test accessing API from process',
+            requirements={'expression-engine': 'jinja'},
+            contributor=self.contributor,
+            type='data:test:api-access',
+            input_schema=[],
+            output_schema=[
+                {'name': 'collection-list', 'type': 'basic:json:'},
+            ],
+            run={
+                'language': 'bash',
+                # NOTE: Resolwe Runtime Utilities are not available in the
+                # resolwe/test:base Docker image so we need to use plain echo
+                'program': r"""
+echo "{\"collection-list\": $(curl $RESOLWE_API_HOST/collection)}"
+"""
+            }
+        )
+        return process
+
+    def setUp(self):
+        """Custom initilization of :class:`~TransactionProcessTestCase`."""
+        # NOTE: We want to skip TransactionProcessTestCase's setUp() and just
+        # call its parent's setUp()
+        # pylint: disable=bad-super-call
+        super(TransactionProcessTestCase, self).setUp()
+
+        self.collection = self.create_collection()
+        self.process = self.create_process()
+
+        self._prepare_files_and_dirs()
+
+    def check_results(self, data):
+        """Check if information obtained for collection via API is correct."""
+        collection_list = Storage.objects.get(data=data).json
+        self.assertEqual(len(collection_list), 1)
+        self.assertEqual(collection_list[0]['slug'], self.collection.slug)
+
+    @unittest.skipUnless(*check_installed('curl'))
+    @with_resolwe_api_host
+    def test_access_api_from_local_executor_process(self):  # pylint: disable=invalid-name
+        """Test if a process running via local executor can access API."""
+        data = self.run_process(self.process.slug)
+        self.check_results(data)
+
+    @with_resolwe_api_host
+    @with_docker_executor
```

Comment:
This doesn't work on macOS/Windows, where docker runs in virtual machine and thus can not access live server running on the local machine.

https://forums.docker.com/t/access-host-not-vm-from-inside-container/11747/39